solver/src/liquidity/offchain_orderbook.rs

```diff
@@ -47,6 +54,262 @@ pub fn normalize_limit_order(order: Order, native_token: WETH9) -> LimitOrder {
 
 impl SettlementHandling<LimitOrder> for OrderSettlementHandler {
     fn encode(&self, executed_amount: U256, encoder: &mut SettlementEncoder) -> Result<()> {
-        encoder.add_trade(self.order.clone(), executed_amount)
+        if self.order.order_creation.buy_token == BUY_ETH_ADDRESS {
+            let interaction = compute_unwrap_interaction(
+                encoder.clearing_prices(),
+                &self.order,
+                executed_amount,
+                self.native_token.clone(),
+            )?;
+
+            encoder.add_trade(self.order.clone(), executed_amount)?;
+
+            encoder.add_unwrap(interaction);
+        } else {
+            encoder.add_trade(self.order.clone(), executed_amount)?;
+        }
+
+        Ok(())
+    }
+}
+
+fn compute_unwrap_interaction(
+    clearing_prices: &HashMap<H160, U256>,
+    order: &Order,
+    executed_amount: U256,
+    weth: WETH9,
+) -> Result<UnwrapWethInteraction> {
+    let sell_price = *clearing_prices
+        .get(&order.order_creation.sell_token)
+        .ok_or_else(|| anyhow!("sell price not available"))?;
+    let buy_price = *clearing_prices
+        .get(&order.order_creation.buy_token)
```

Comment:
I think this might be the issue you mentioned, where its trying to get the price of `0xeeeeeee...eeee` instead of `WETH.address()` which is what the solver outputs.

I think that, in order to fix this, we need to change the encoder to add native token address and price to the list.